### PR TITLE
Unpin uber-go/tally version

### PR DESCRIPTION
**Pull request by yiranwang52**

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ ignore:
 - go/doc
 import:
 - package: github.com/uber-go/tally
-  version: v2.1.0
+  version: ^v2.1.0
```
Comment (prashantv) [inline glide.yaml]: tally is on version 3, so might be better to do `^3`

Comment (yiranwang52) [reply glide.yaml]: okay.

File: glide.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ ignore:
 - go/doc
 import:
 - package: github.com/uber-go/tally
-  version: ^v2.1.0
+  version: ^v3
```
Comment (prashantv) [inline glide.yaml]: you need to run `glide up` after changing it, and i think you can do just `^3`

Comment (yiranwang52) [reply glide.yaml]: I ran `glide up`, but github hides `glide.lock` changes by default.
Changing `^v3` to `^3`.

**Pull request merged**